scripts/distribution/macOS-package/build.sh

```diff
@@ -0,0 +1,373 @@
+function compileConsensus() {
+    cd "$consensusDir"
+    logInfo "Building Consensus..."
+    stack build
+    logInfo "Done"
+}
+
+# Compile the node and collector using dynamic linking.
+function compileNodeAndCollector() {
+    cd "$nodeDir"
+    logInfo "Building Node and Collector..."
+    cargo build --bin concordium-node --bin node-collector --features collector --release
+    logInfo "Done"
+}
+
+# Compile the installer plugin using xcodebuild.
+# The plugin is located in 'NodeConfigurationInstallerPlugin'.
+function compileInstallerPlugin() {
+    logInfo "Building installer plugin..."
+    xcodebuild -project "$macPackageDir/NodeConfigurationInstallerPlugin/NodeConfigurationInstallerPlugin.xcodeproj" > /dev/null
+    logInfo "Done"
+}
+
+# Compile consensus, node, collector, and the installer plugin.
+function compile() {
+    compileConsensus
+    compileNodeAndCollector
+    compileInstallerPlugin
+}
+
+# Copy the needed compiled files from the installer plugin to the build folder.
+function copyInstallerPluginData() {
+    logInfo "Copying installer plugin data to build folder"
+    cp -r "$macPackageDir/NodeConfigurationInstallerPlugin/build/Release/NodeConfigurationInstallerPlugin.bundle" "$buildDir/plugins"
+    cp "$macPackageDir/NodeConfigurationInstallerPlugin/NodeConfigurationInstallerPlugin/InstallerSections.plist" "$buildDir/plugins"
+    logInfo "Done"
+}
+
+# Copy the node and collector binaries to the build folder.
+function copyNodeBinaries() {
+    logInfo "Copy concordium-node and node-collector binaries to '$payloadDir/Library/Concordium Node/'.."
+    cp "$nodeDir/target/release/concordium-node" "$payloadDir/Library/Concordium Node"
+    cp "$nodeDir/target/release/node-collector" "$payloadDir/Library/Concordium Node"
+    logInfo "Done"
+}
+
+# Copy the compiled items (binaries and supporting data) to the build folder.
+function copyCompiledItemsToBuildDir() {
+    copyNodeBinaries
+    copyInstallerPluginData
+}
+
+# Get the tool dylibbundler, which is used to recursively find and
+# bundle the needed dylibs for a binary.
+# Checks whether the tool is already installed and then potentially skips the build step.
+function getDylibbundler() {
+    logInfo "Getting dylibbundler..."
+
+    if test -f "$macdylibbundlerDir/dylibbundler"
+    then
+        logInfo "Done (skipped: already exists)"
+    else
+        logInfo " -- Downloading..."
+        mkdir "$toolsDir"
+        cd "$macPackageDir"
+        curl -sSL "https://github.com/auriamg/macdylibbundler/archive/refs/tags/1.0.0.zip" > "$toolsDir/dylibbundler.zip" \
+                    && logInfo " -- Unzipping..." \
+                    && cd "$toolsDir" \
+                    && unzip "dylibbundler.zip" \
+                    && logInfo " -- Building..." \
+                    && cd "$macdylibbundlerDir" \
+                    && make
+        logInfo "Done"
+    fi
+}
+
+# Use dylibbundler to recursively find and bundle the dylibs for node and collector.
```

Comment:
I thought we agreed not to use dylibbundler, but to do the bundling ourselves?

Reply:
Ales and I arrived at the conclusion that it is ok for now, as we have a fixed resolver for Stack. This means that builds will use the exact same versions of the dependencies. Installers built on different machines might have different hashes, though, as the dependencies are built on different machines and GHC doesn't promise that level of reproducibility.